src/main/kotlin/second_pass/SymbolResolverVisitor.kt

```diff
@@ -0,0 +1,103 @@
+package second_pass
+
+import com.github.javaparser.ast.Node
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration
+import com.github.javaparser.ast.body.ConstructorDeclaration
+import com.github.javaparser.ast.body.MethodDeclaration
+import com.github.javaparser.ast.body.VariableDeclarator
+import com.github.javaparser.ast.expr.FieldAccessExpr
+import com.github.javaparser.ast.expr.MethodCallExpr
+import com.github.javaparser.ast.visitor.VoidVisitorAdapter
+import first_pass.node_parser.FieldDeclarationParser
+import second_pass.context.ClassContext
+import second_pass.context.MemberContext
+import second_pass.context.VisitorContext
+import hierarchy.HierarchyGraph
+import second_pass.resolver.SymbolResolver
+import second_pass.signature.MemberDeclarationSignature
+import second_pass.signature.MemberSignature
+import util.ResolverVisitorResource
+
+class SymbolResolverVisitor : VoidVisitorAdapter<ClassContext>() {
+
+    private lateinit var visitorContext: VisitorContext
+    private lateinit var hierarchyVisitor: HierarchyVisitor
+    private lateinit var resolver: SymbolResolver
+
+    private fun initializeDataStructures(hierarchyGraph: HierarchyGraph) {
+        visitorContext = VisitorContext()
+        hierarchyVisitor = HierarchyVisitor(visitorContext, hierarchyGraph)
+        resolver = SymbolResolver(visitorContext, hierarchyGraph)
+    }
+
+// Resolving methods
+    fun resolveResources(hierarchyGraph: HierarchyGraph, resources: List<ResolverVisitorResource>) {
```

Comment:
Regarding _ResolverVisitorResource_ - what would happen to the design if this pairing of the compilation unit and associated cadetclass did not exist? How would it change?

Reply:
ResolverVisitorResource does not exist anymore, it has been replaced with 2 parameters (which were contained within this dto anyways).